python/src/iceberg/utils/iceberg_base_model.py

```diff
@@ -0,0 +1,29 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from pydantic import BaseModel
+
+
+class IcebergBaseModel(BaseModel):
+    class Config:
+        allow_population_by_field_name = True
+
+    def dict(self, exclude_none=True, **kwargs):
+        return super().dict(exclude_none=exclude_none, **kwargs)
+
+    def json(self, exclude_none=True, by_alias=True, **kwargs):
+        return super().json(exclude_none=exclude_none, by_alias=True, **kwargs)
```

Comment:
What is this implementation changing? Wouldn't the superclass's `json` method be called with these args? Is this excluding allowed arguments?

Reply:
In the superclass, the `by_alias` is set to False by default. But for Iceberg it is better to set this to `True` all the time because we rely on aliases when we convert JSON we always want to use the aliases (`schema-id` instead of `schema_id`, since `-` is an operator in Python). The same goes for excluding non-values in the case of the doc element of the NestedField that we don't want to include if it is `None`.

By overriding this function, we set these values by default to True.

Reply:
Thanks for explaining! That makes sense. Can you make sure that this is clear in a comment? That way future readers will also know.

Also, should the `by_alias` passed to the method be passed to the `super().json(...)` call? Right now it ignores it.

Reply:
I missed your previous comments, sorry about that!

> Also, should the by_alias passed to the method be passed to the super().json(...) call? Right now it ignores it.

Thanks! That was a copy-paste error.

> I've added the following comment:

```
This class extends the Pydantic BaseModel to set default values by overriding them.

This is because we always want to set by_alias to True. In Python, the dash can't
be used in variable names, and this is used throughout the Iceberg spec.

The same goes for exclude_none, if a field is None we want to omit it from
serialization, for example, the doc attribute on the NestedField object.
Default non-null values will be serialized.

This is recommended by Pydantic:
https://pydantic-docs.helpmanual.io/usage/model_config/#change-behaviour-globally
```